elf_ctx.c

```diff
@@ -164,7 +164,7 @@ void elf_ctx_init(struct ukarch_ctx *ctx, struct elf_prog *prog,
 	 * - environment variables (plus extra NULL pointer);
 	 * - arguments (plus extra NULL pointer and argument count)
 	 */
-	stackvalc = AUXV_VALUES_PER_ENTRY * ARRAY_SIZE(auxv) + (envc+1) + ((argc-1)+2);
+	stackvalc = AUXV_VALUES_PER_ENTRY * ARRAY_SIZE(auxv) + (envc+1) + (argc+2);
```

Comment:
Why do we need to do `+2` to `argc`?

Reply:
Ah right, I understood now. More readable might be:
`(argc + 1) + 1`. The extra `+1` is the space needed for putting `argc` on the stack: http://articles.manugarg.com/aboutelfauxiliaryvectors.html
:thinking: If you update the computation with the PR #9 you could directly in that PR add another `sizeof(long)` to match the data type with pushed `argc` (line 200).

Reply:
I have addressed this in the other PR.

Regarding the mixing of data types in all the calculations above -- well that is because at the end of the day, each individual value pushed on the stack, no matter the type, takes `sizeof(void*) = word size` bytes. If we add support for an architecture where `sizeof(long) != sizeof(void*)` (eg: Windows x64 comes to mind, but that is not an architecture :smile:), we might have to revisit these calculations.